### PR TITLE
Feature/uppsf 3069 sync ontology

### PR DESCRIPTION
**Pull request by ivan-p-nikolov**
# Description

## What

- Sync ontology package structure between the service and cm-graph-ontology
- Rename concept models to have the same names as in the ontology lib
- Sync concept model fields

## Why

https://financialtimes.atlassian.net/browse/UPPSF-3069

## Anything, in particular, you'd like to highlight to reviewers

Note that the models between the service and the lib are still different.
This is because they use different styles of transforming between the old and new models.
Aggy relays heavily on json Marshal/Unmarshal, where the lib has custom interface to switch between models.

I personally like the marshal approach more, as it allows us to keep the old flat json messaging structure for concepts. 
And so I plan to copy this approach to the ontology lib.

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [ ] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

File: ontology/aggregate/aggregate.go [hunk 1]
```diff
@@ -15,7 +15,8 @@ import (
 func CreateAggregateConcept(primary ontology.NewConcept, others []ontology.NewConcept) ontology.NewAggregatedConcept {
 	var scopeNoteOptions = map[string][]string{}
 	concordedConcept := ontology.NewAggregatedConcept{}
-	concordedConcept.Fields = map[string]interface{}{} // initialise Fields to be able to safely access it later
+	concordedConcept.Properties = map[string]interface{}{}    // initialise Properties to be able to safely access it later
+	concordedConcept.Relationships = map[string]interface{}{} // initialise Properties to be able to safely access it later
```
Comment (ManoelMilchev) [inline ontology/aggregate/aggregate.go:19]: `nitpick` : // initialise Properties -> Initialise Relationships

File: ontology/concordedConcept.go [hunk 2]
```diff
@@ -12,13 +12,17 @@ type NewAggregatedConcept struct {
 
 type RequiredConcordedFields struct {
 	// Required fields
-	PrefUUID  string `json:"prefUUID,omitempty"`
-	PrefLabel string `json:"prefLabel,omitempty"`
-	Type      string `json:"type,omitempty"`
+	PrefUUID       string `json:"prefUUID"`
+	PrefLabel      string `json:"prefLabel"`
+	Type           string `json:"type"`
+	AggregatedHash string `json:"aggregateHash,omitempty"`
```
Comment (ManoelMilchev) [inline ontology/concordedConcept.go:18]: `question`: How is this field `Required` but have `omitempty` ?

**Review by ManoelMilchev — APPROVED** [review]
LGTM! Just a few questions.

`praise`: I really like how you separate the commits! Trying to do this myself for some time now. :muscle:

File: ontology/aggregate/aggregate.go (hunk 1 above)
Comment (ManoelMilchev) [reply ontology/aggregate/aggregate.go:19]: Ignoring this comment as the next PR resolves it.

**Review by ivanruski — APPROVED** [review]

File: ontology/concordedConcept.go (hunk 2 above)
Comment (ivan-p-nikolov) [reply ontology/concordedConcept.go:18]: It is all a setup for the ontology extraction.
`Required*Fields` are all the fields that should be always present on a Concept node no matter what. `Additional*Fields` are optional for the nodes.

The json serialisation is separate from that. The json structure is our messaging format. Meaning that we don't to force some fields to be present.

Comment (ManoelMilchev) [reply ontology/concordedConcept.go:18]: Thank you!

**Pull request merged**